docs/GITHUB_ACTIONS.md

```diff
@@ -169,23 +291,116 @@ The action only runs when:
 This function looks complex. @kodelet can you refactor this?
 ```
 
-## Permissions Required
+## Permissions
 
 The action requires the following GitHub permissions:
 
+**For Auth Gateway (Default & Recommended):**
+```yaml
+permissions:
+  id-token: write
+  issues: read
+  pull-requests: read
+  contents: read
+```
+
+**For Manual Override:**
 ```yaml
 permissions:
   issues: write          # Comment on issues
   pull-requests: write   # Comment on PRs and create PRs
   contents: write        # Push commits and create branches
 ```
 
-## Security Considerations
+## GitHub Authentication Token Considerations
+
+### 1. Auth Gateway (Default & Recommended)
+
+The action uses GitHub's OIDC to authenticate with an auth gateway service that provides elevated permissions automatically. This is the **most secure and convenient** option.
+
+**Benefits:**
+- No token management required
+- Automatic permission elevation
+- Secure credential handling
+- Simplified workflow setup
+
+**Setup:**
+```yaml
+permissions:
+  id-token: write
+  issues: read
+  pull-requests: read
+  contents: read
+
+steps:
+  - uses: jingkaihe/kodelet-action@v0.1.7-alpha
+    with:
+      anthropic-api-key: ${{ secrets.ANTHROPIC_API_KEY }}
+      # github-token is automatically handled
+```
+
+### 2. Standard GitHub Token
+
+Uses the default `GITHUB_TOKEN` provided by GitHub Actions. **Limited permissions** - can only comment, cannot create PRs or push commits.
+
+**Limitations:**
+- Cannot create pull requests
+- Cannot push commits to new branches
+- Can only post comments
+
+**Setup:**
+```yaml
+permissions:
+  issues: write
+  pull-requests: write
+
+steps:
+  - uses: jingkaihe/kodelet-action@v0.1.7-alpha
+    with:
+      anthropic-api-key: ${{ secrets.ANTHROPIC_API_KEY }}
+      github-token: ${{ secrets.GITHUB_TOKEN }}
+```
+
+### 3. Personal Access Token (PAT)
+
+Use a fine-grained or classic PAT for full repository access.
+
+**Requirements:**
+- Create PAT with necessary permissions
+- Store in repository secrets
+- More token management overhead
+
+**Setup:**
+```yaml
+steps:
+  - uses: jingkaihe/kodelet-action@v0.1.7-alpha
+    with:
+      anthropic-api-key: ${{ secrets.ANTHROPIC_API_KEY }}
+      github-token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+```
+
+### 4. Custom Auth Gateway
+
+Advanced users can configure their own auth gateway endpoint:
```

Comment:
clarify "for the purpose of enterprise deployment or additonal security compliance requirement". and mention "the source code of the Github Actions auth gateway to be open sourced soon